### ISSUE TITLE
Generate the list of required CLI commands to deploy the infrastructure at build time

### ISSUE THREAD

**Issue by mauroservienti** [Particular/NServiceBus.AmazonSQS#1176]
The SQS transport comes with a CLI that helps in [creating the required resources](https://docs.particular.net/transports/sqs/operations-scripting#create-resources).

An improvement to the CLI experience would be that, when compiling the source code of an endpoint using the SQS transport, an artifact containing the list of the required commands to deploy the infrastructure is output in the build directory. Similarly to what SQL Persistence does for [SQL scripts](https://docs.particular.net/persistence/sql/controlling-script-generation).

**Comment by DavidBoike** [comment]
While SQL Persistence is generating SQL tables, it might be quite a bit more difficult to generate metadata such as what message types to subscribe to at compile time.

An alternative would be for installers in the package to generate log statements that contain the required CLI commands for what the Installers are doing at that moment.

Then a developer would be able to debug locally, but then copy the log statements into a script.

**Comment by twonder** [comment]
I would agree that after looking at the CLI commands, it wasn't super obvious what ones I would need to run in order to set it up properly.

One thing I would point out is it is generally best practice to use CloudFormation to deploy and manage resources together in "stacks", instead of managing them individually.  Since the construction of these commands in manual at the moment, I was planning on using the CDK to create a CloudFormation template with the required resources.

That being said, I wonder if a better approach to provisioning these resources is to use a .NET CDK app to dynamically create a CloudFormation stack and all of the required queues, topics and permissions by scanning the necessary assemblies.

**Comment by danielmarbach** [comment]
One other thing we briefly discussed today was that 8.1 will introduce a [dedicated installer API](https://github.com/Particular/NServiceBus/pull/6563). With that, it would also be possible to run the endpoint in sort of installer mode in which we could dump out a metamodel of the topology that can be transformed into CloudFormation or CLI scripts.

**Comment by johnsimons** [comment]
This is related to https://github.com/Particular/NServiceBus/issues/7370.
